unfetter-discover/unfetter-ui
# Issue 883 (Capability Selector for Baselines wizard)

Fixes unfetter-discover/unfetter#883
Supports unfetter-discover/unfetter#956

Capability selector, i.e. the component which appears after the capability group selection step of the baselines wizard.  This component relies on a current capability group to which capabilities are associated, and writes out the selected capabilities to the baseline store.  Handling of each selected capability group is handled at the wizard level - this handling will be in an upcoming issue.

## To test:

1. Pull this PR
2. Make sure these lines are uncommented in unfetter/docker-compose.yml:

```- /tmp/examples/unfetter-stix/assessments3/capabilities.stix.json```
```- /tmp/examples/unfetter-stix/assessments3/categories.stix.json```

3. Update line 170 of wizard.component.ts to set the capability selector as the first page of the wizard:

```this.openedSidePanel = 'categories';```

4. Restart unfetter-ui:  ``docker-compose restart unfetter-ui``
5. Open up Baselines in the UI and create a new one
6. Enter title, description, and org (note that title does not update correctly - will be fixed)
7. A random capability group will be used as the current capability group and should be the title of the wizard page
8. Select one or more capabilities
9. Click Continue
10. With Redux Tools, confirm the ``baseline`` store has the selected capabilities saved in ``baselineCapabilities``.
